hoot-core/src/main/cpp/hoot/core/io/OsmXmlWriter.cpp

```diff
@@ -300,44 +301,41 @@ void OsmXmlWriter::_writeTags(const ConstElementPtr& element)
 
   const ElementType type = elementClone->getElementType();
   assert(type != ElementType::Unknown);
-  const Tags& tags = elementClone->getTags();
-
-  for (Tags::const_iterator it = tags.constBegin(); it != tags.constEnd(); ++it)
-  {
-    QString key = it.key();
-    QString val = it.value().trimmed();
-    if (val.isEmpty() == false)
-    {
-      _writer->writeStartElement("tag");
-      LOG_VART(key);
-      _writer->writeAttribute("k", removeInvalidCharacters(key));
-      LOG_VART(val);
-      _writer->writeAttribute("v", removeInvalidCharacters(val));
-      _writer->writeEndElement();
-    }
-  }
+  Tags& tags = elementClone->getTags();
```

Comment:
Since the element is cloned, updating the reference to the tags is ok because it will be thrown away after writing.  This allows us to add other tags and sort or not sort below.